### ISSUE TITLE
After 6 epochs interim results turn out to be abnormal.

### ISSUE THREAD

**Issue by goks** [leftthomas/ImageDeblurring#16]
![6_181](https://user-images.githubusercontent.com/11703678/37447657-0e69f136-2849-11e8-9458-561edc8e8c9f.png)
![6_211](https://user-images.githubusercontent.com/11703678/37447658-0eb8c63a-2849-11e8-8993-a7cbbdaa1287.png)
![6_241](https://user-images.githubusercontent.com/11703678/37447659-0ee6c382-2849-11e8-9917-8ca2bec92382.png)
![6_271](https://user-images.githubusercontent.com/11703678/37447660-0f41c264-2849-11e8-9447-a28c0bf680b5.png)
![6_301](https://user-images.githubusercontent.com/11703678/37447662-0f9154d2-2849-11e8-99ce-9f4f51fd52f6.png)
![6_331](https://user-images.githubusercontent.com/11703678/37447663-0fbfb336-2849-11e8-960f-4798f7236b25.png)

**Comment by goks** [comment]
6_181, 6_211, 6_241, 6_271, 6_301, 6_331 are the exact epoch and batch numbers corresponding to the images.

**Comment by percivalstr** [comment]
wired

**Comment by Imagicguy** [comment]
I also encountered this trouble after 2 epochs. and the d_on_g_loss =0. Did you solve this problem? I would appreciate your reply!

**Comment by Imagicguy** [comment]
I think it might be the" overfitting " which leaded to this problem.I tried to change the channel rate from 64 to 16,it worked .And I will try to enlarge the dataset (maybe crop the images),add regularization into the 
Convolution2D layer .

**Comment by ghost** [comment]
@goks @Imagicguy 
Hey!
First i have to say i am very new to GAN networks, as well as keras and tensorflow!
I have the problem that i can not run this code!? If a start it with "python main.py" it cannot find the data/data.h5 because its not built yet. So i tried to run data_utils first and than tha main.py but it dosent work. Could you guys give me a hint how to get his code running? 
I really would appreciate it!

**Comment by leftthomas** [comment]
@mlauskoser you can try to change the channel rate from 64 to 16

**Comment by HPL123** [comment]
@goks I also encountered this trouble same with you. Did you reslove it?

**Comment by machengcheng2016** [comment]
Thanks for you reply! Seems that I met the same situation as you referred here. But **why changing channel rate can avoid this abnormal situation?** Is there a explanation for this? Besides, **what does "d_on_g_loss = 0 " mean?** Since we may fail to trust the training process as it converges to a wrong solution. Thank you again!

**Comment by machengcheng2016** [comment]
@leftthomas 
Thanks for you reply! Seems that I met the same situation as you referred here. **But why does changing channel rate work?** Is there a explanation for this? **Besides, what does "d_on_g_loss = 0 " mean, and how can we avoid this abnormal situation?** Since we may fail to trust the training process as it converges to a wrong solution. Thank you again!

**Comment by Imagicguy** [comment]
Well，d_loss=0 means the discriminator thought the generator had already studied all the features needed in the deblur process.It is the reason of overfitting.
		

		获取 Outlook for iOS
	





On Thu, Jun 21, 2018 at 7:12 PM +0800, "machengcheng2016" <notifications@github.com> wrote:












Thanks for you reply! Seems that I met the same situation as you referred here. But why changing channel rate can avoid this abnormal situation? Is there a explanation for this? Besides, what does "d_on_g_loss = 0 " mean? Since we may fail to trust the training process as it converges to a wrong solution. Thank you again!

**Comment by Imagicguy** [comment]
You can avoid overfitting by dropout,data augment and regularization.
		

		获取 Outlook for iOS
	收件人: machengcheng2016 <notifications@github.com>
Sent: 星期四, 六月 21, 2018 19:17
收件人: leftthomas/ImageDeblurring
抄送: Imagicguy; Mention
主题: Re: [leftthomas/ImageDeblurring] After 6 epochs interim results turn out to be abnormal. (#16) 

@leftthomas
Thanks for you reply! Seems that I met the same situation as you referred here. But why does changing channel rate work? Is there a explanation for this? Besides, what does "d_on_g_loss = 0 " mean, and how can we avoid this abnormal situation? Since we may fail to trust the training process as it converges to a wrong solution. Thank you again!

**Comment by machengcheng2016** [comment]
@Imagicguy @leftthomas 
Thank you for your kind reply!
Personally, I think the reason for over-fitting is that the architecture of Discriminator network may be inappropriate. Since I've checked the original code implemented in torch written by author, the discriminator network will output a label map, instead of a single score, a single value. There seems a little difference on the definition of PatchGAN between this code and the paper author's code. You think?

**Comment by fengwang** [comment]
@goks  I interpret this as typical distortion introduced by batch normalization layers.  One way is to increase the batch size with more GPUs; The other way is to continue the training, hoping these distortion will go away if you are fortunate enough.
